### PR TITLE
Add GQA enhancements

### PR DESCRIPTION
**Pull request by Aswinmcw**
Introduce softcapping, RMSNorm support, and exp and test coverage

- Updated `gqa_attention_torch` to include parameters for attention logit softcapping, QK normalization weights, and custom scaling.
- Added RMSNorm functionality for Q and K heads.
- Expanded test suite in `test_gqa.py` to cover new features, including causal masking and RoPE variations.
- Introduced GQA-specific profiles in `helpers.py` for full and tiny model configurations, enhancing testing capabilities.

### Ticket
Link to Github Issue

### Problem description
Provide context for the problem.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist

- [ ] [Sanity tests](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:Aswinmcw/gqa_testing_model)
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:Aswinmcw/gqa_testing_model)
- [ ] [cpp-unit-tests](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:Aswinmcw/gqa_testing_model)
- [ ] New/Existing tests provide coverage for changes
- [ ] (Optional) Ran [clang-tidy code analysis](https://github.com/tenstorrent/tt-metal/actions/workflows/code-analysis.yaml) on the PR branch to catch linting errors early

#### Model tests

If your changes cover model-related code, you should run tests corresponding to affected models and platforms (Single card, T3K, Galaxy). "Choose your pipeline" workflows facilitate running multiple kinds of tests in a single run. Each offers `models-mandatory` and `models-extended` presets.
The former includes a minimal set of tests, to be run always. The latter extends that with additional ones - use your best judgement in deciding which is the most appropriate for your PR.

- [ ] [(Single) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:Aswinmcw/gqa_testing_model)
  - [ ] `models-mandatory` preset (runs: [Device perf regressions](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) and [Frequent model and ttnn tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) tests)
  - [ ] other selection - specify runs

- [ ] [(T3K) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:Aswinmcw/gqa_testing_model)
  - [ ] `models-mandatory` preset (runs: [Unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-model-perf-tests.yaml) tests)
  - [ ] other selection - specify runs

- [ ] [(Galaxy) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:Aswinmcw/gqa_testing_model)
  - [ ] `models-mandatory` preset (runs: [Quick tests](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml))
  - [ ] `models-extended` preset (runs: the mandatory tests, plus [Demo](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) and [Model perf](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-perf-tests.yaml) tests)
  - [ ] other selection - specify runs

### CODE REVIEW

File: models/demos/deepseek_v3_b1/tests/block_tests/gqa.py [hunk 1]
```diff
@@ -38,16 +44,26 @@ def gqa_attention_torch(
     rope_variant: "standard" (Llama/Qwen) or "glm4" (interleaved, partial)
     attention_mask: None or additive [b,1,q,kv]
     past_key_value: None or (past_k, past_v) each [b, kv_heads, past_seq, hd]
+    attn_logit_softcapping: if set, tanh-softcap logits to this value (Grok-2)
+    qk_norm_weights: None or (q_norm_weight, k_norm_weight) for per-head RMSNorm
+    qk_norm_eps: epsilon for QK RMSNorm
+    scaling: custom Q·K scaling factor (defaults to 1/sqrt(head_dim))
```
Comment (itarabanTT) [inline models/demos/deepseek_v3_b1/tests/block_tests/gqa.py]: Does this `scaling` parameter is used in any model?

File: models/demos/deepseek_v3_b1/tests/block_tests/gqa.py [hunk 2]
```diff
@@ -38,16 +44,26 @@ def gqa_attention_torch(
     rope_variant: "standard" (Llama/Qwen) or "glm4" (interleaved, partial)
     attention_mask: None or additive [b,1,q,kv]
     past_key_value: None or (past_k, past_v) each [b, kv_heads, past_seq, hd]
+    attn_logit_softcapping: if set, tanh-softcap logits to this value (Grok-2)
```
Comment (itarabanTT) [inline models/demos/deepseek_v3_b1/tests/block_tests/gqa.py]: Are you sure that this parameters is used in Grok-2?
